java/compression/src/test/java/org/apache/arrow/compression/TestArrowReaderWriterWithCompression.java

```diff
@@ -79,11 +119,130 @@ public void testArrowFileZstdRoundTrip() throws Exception {
            new ArrowFileReader(new ByteArrayReadableSeekableByteChannel(out.toByteArray()),
              allocator, NoCompressionCodec.Factory.INSTANCE)) {
       Assert.assertEquals(1, reader.getRecordBlocks().size());
+      Exception exception = Assert.assertThrows(IllegalArgumentException.class, () -> reader.loadNextBatch());
+      String expectedMessage = "Please add arrow-compression module to use CommonsCompressionFactory for ZSTD";
+      Assert.assertEquals(expectedMessage, exception.getMessage());
+    }
+  }
+
+  @Test
+  public void testArrowFileZstdRoundTripWithDictionary() throws Exception {
+    // Prepare sample data
+    DictionaryProvider.MapDictionaryProvider provider = new DictionaryProvider.MapDictionaryProvider();
+    provider.put(dictionary1);
+
+    final BufferAllocator allocator = new RootAllocator(Integer.MAX_VALUE);
```

Comment:
I think you can get rid of these in favor of this.allocator if you keep the `@Before/@After` functions

Reply:
Yes, I agree. I refactored the code to use this approach.